lighthouse-core/lib/i18n/README.md

```diff
@@ -0,0 +1,305 @@
+# UIStrings Syntax
+
+## ICU Syntax
+
+More about [complex ICU](http://userguide.icu-project.org/formatparse/messages)
+formatting.
+
+### Selects
+
+A select ICU message is used when the message should select a sub-message based
+on the value of a variable `pronoun` in this case.  This is often used for gender
+based selections, but can be used for any enum.  Lighthouse does not use selects
+very often.
+
+```javascript
+displayValue: `{pronoun, select,
+  male {He programmed the link.}
+  female {She programmed the link.}
+  other {They programmed the link}
+  }`,
+```
+
+### Ordinals (Numeric Selects)
+
+An ordinal ICU message is used when the message contains "plurals", wherein a
+sub-message would need to be selected from a list of messages depending on the
+value of `itemCount` (in this example).  They are a flavor of "Selects" that
+have a unique syntax.
+
+```javascript
+displayValue: `{itemCount, plural,
+  =1 {1 link found}
+  other {# links found}
+  }`,
+```
+
+### Primitive Formatting
+
+This include number formatting like `{timeInMs, number, milliseconds}` and
+string replacements like `{some_name}`.
+
+## message.json Syntax
+
+### Why we use message.json
+
+It is the
+[Chrome Standard](https://developer.chrome.com/extensions/i18n-messages) for
+i18n messages.
+
+### Parts of a message.json message
+
+(From the Chrome Docs)
+
+```json
+{
+  "name": {
+    "message": "Message text, with optional placeholders.",
+    "description": "Translator-aimed description of the message.",
+    "placeholders": {
+      "placeholder_name": {
+        "content": "A string to be placed within the message.",
+        "example": "Translator-aimed example of the placeholder string."
+      },
+    }
+  }
+}
+```
+
+## Our message system
+
+We use UIStrings & i18n.js to extract strings from individual js files into
+locale.json files. This allows us to keep strings close to the code in which
+they are used so that developers can easily understand their context. It also
+comes with its own syntax.
+
+UIStrings are defined in a simple Object with properties defining the strings.
+
+*   simple string named "title", with no description:
+
+    ```javascript
+    const UIStrings = {
+      title: 'Minify CSS',
+    };
+    ```
+
+*   simple string, with a description above property as a comment:
+
+    ```javascript
+    const UIStrings = {
+      /** Imperative title of a Lighthouse audit that ... */
+      title: 'Minify CSS',
+    };
+    ```
+
+### Markdown
+
+Strings can also contain some limited markdown, that will be rendered in the
+report.
+
+*   Code block.
+
+    To format some text as code it should be contained in `backticks`. These
```

Comment:
```suggestion
    To format some text as code it should be contained in `backticks`. Any text within the backticks will not be translated.
```

Reply:
this is a convenient way to do this, but are we sure we want to limit ourselves from ever translating text inside backticks? Or is there still a way to do that with replacements somehow

Reply:
Good point, I think for now it's okay because its used for very short things, but in future, maybe.  I don't know why code _would_ be translated, but :man_shrugging: